contracts/smart-account/modules/SessionKeyManagers/SessionKeyManagerHybrid.sol

```diff
@@ -0,0 +1,744 @@
+         * 0xe                  | 0x14              | Session Validation Module Address
+         * 0x22                 | --                | abi.encode(sessionKeyData, sessionEnableData,
+         *                      |                   |   sessionEnableSignature, sessionKeySignature)
+         */
+        assembly ("memory-safe") {
+            let offset := add(_moduleSignature.offset, 0x1)
+
+            sessionKeyIndex := shr(248, calldataload(offset))
+            offset := add(offset, 0x1)
+
+            validUntil := shr(208, calldataload(offset))
+            offset := add(offset, 0x6)
+
+            validAfter := shr(208, calldataload(offset))
+            offset := add(offset, 0x6)
+
+            sessionValidationModule := shr(96, calldataload(offset))
+            offset := add(offset, 0x14)
+
+            let baseOffset := offset
+            let dataPointer := add(baseOffset, calldataload(offset))
+
+            sessionKeyData.offset := add(dataPointer, 0x20)
+            sessionKeyData.length := calldataload(dataPointer)
+            offset := add(offset, 0x20)
+
+            dataPointer := add(baseOffset, calldataload(offset))
+            sessionEnableData.offset := add(dataPointer, 0x20)
+            sessionEnableData.length := calldataload(dataPointer)
+            offset := add(offset, 0x20)
+
+            dataPointer := add(baseOffset, calldataload(offset))
+            sessionEnableSignature.offset := add(dataPointer, 0x20)
+            sessionEnableSignature.length := calldataload(dataPointer)
+            offset := add(offset, 0x20)
+
+            dataPointer := add(baseOffset, calldataload(offset))
+            sessionKeySignature.offset := add(dataPointer, 0x20)
+            sessionKeySignature.length := calldataload(dataPointer)
+        }
+    }
+
+    function _parseSessionDataPreEnabledSignatureSingleCall(
+        bytes calldata _moduleSignature
+    )
+        internal
+        pure
+        returns (bytes32 sessionDataDigest_, bytes calldata sessionKeySignature)
+    {
+        /*
+         * Session Data Pre Enabled Signature Layout
+         * Offset (in bytes)    | Length (in bytes) | Contents
+         * 0x0                  | 0x1               | Is Session Enable Transaction Flag
+         * 0x1                  | --                | abi.encode(bytes32 sessionDataDigest, sessionKeySignature)
+         */
+        assembly ("memory-safe") {
+            let offset := add(_moduleSignature.offset, 0x1)
+            let baseOffset := offset
+
+            sessionDataDigest_ := calldataload(offset)
+            offset := add(offset, 0x20)
+
+            let dataPointer := add(baseOffset, calldataload(offset))
+            sessionKeySignature.offset := add(dataPointer, 0x20)
+            sessionKeySignature.length := calldataload(dataPointer)
+        }
+    }
+
+    /***************************** Batch Call Handler ***********************************/
+
+    function _validateUserOpBatchExecute(
```

Comment:
If it is possible w/o introducing gas overhead, I would suggest refactoring this function to make it shorter and easier to read.

Reply:
Refactored the handling logic based on `isSessionEnableTransaction` into separate handler functions